src/mdns.c

```diff
@@ -112,31 +121,75 @@ mdns_list_interfaces(multicast_if** pp_intfs, size_t* p_nb_intf, int ai_family)
                 return (MDNS_ERROR);
         }
         *pp_intfs = intfs = malloc(sizeof(*intfs) * nb_if);
-        if (intfs == NULL) {
-                freeifaddrs(ifs);
-                return (MDNS_ERROR);
-        }
+        if (intfs == NULL) 
+                goto error;
+
+        *pp_ip_addresses = ip_addresses = malloc(sizeof(char*) * nb_if);
+
+        if (ip_addresses == NULL) 
+                goto error;
+
         for (c = ifs; c != NULL; c = c->ifa_next) {
                 if (c->ifa_addr == NULL ||
                     c->ifa_addr->sa_family != ai_family ||
                     !mdns_is_interface_valuable(c))
                         continue;
                 memcpy(intfs, c->ifa_addr, sizeof(*intfs));
+                
+                char address_buf[INET6_ADDRSTRLEN];
+
+                const char *ip_address;
+                if (ai_family == AF_INET)
+                {
+                        struct sockaddr_in *addr_in = (struct sockaddr_in *)c->ifa_addr;
+                        ip_address = inet_ntop( ai_family, &addr_in->sin_addr, 
+                                                    address_buf, sizeof( address_buf ) );
+                }
+                else
+                {
+                        struct sockaddr_in6 *addr_in = (struct sockaddr_in6 *)c->ifa_addr;
+                        ip_address = inet_ntop( ai_family, &addr_in->sin6_addr, 
+                                                    address_buf, sizeof( address_buf ) );
+                }
+
+                *ip_addresses = strdup(ip_address);
+                if (*ip_addresses == NULL)
+                        goto error;
+                ip_addresses++;
                 intfs++;
         }
         freeifaddrs(ifs);
         *p_nb_intf = nb_if;
         return (0);
+error:
+        while (ip_addresses != *pp_ip_addresses)
+        {
+                free(*ip_addresses);
+                ip_addresses--;
+        }
+        free(*pp_ip_addresses);
+        free(*pp_intfs);
+        freeifaddrs(ifs);
+        return (MDNS_ERROR);
 }
 #else
 static size_t
-mdns_list_interfaces(multicast_if** pp_intfs, size_t* p_nb_intf, int ai_family)
+mdns_list_interfaces(multicast_if** pp_intfs, char ***pp_ip_addresses, size_t* p_nb_intf, int ai_family)
 {
         multicast_if *intfs;
         *pp_intfs = intfs = malloc(sizeof(*intfs));
         if (intfs == NULL)
                 return (MDNS_ERROR);
         memset(intfs, 0, sizeof(*intfs));
+
+        *pp_ip_addresses = malloc(sizeof(char*));
+        if (pp_ip_addresses == NULL)
+        {
+                free(*pp_intfs);
+                return (MDNS_ERROR);
+        }
+        memset(pp_ip_addresses, 0, sizeof(char*));
```

Comment:
I might be missing something, but couldn't you just store `NULL` in `*pp_ip_addresses`?

Reply:
Yes, I just followed what was consistent with the lines above (line 183)

Reply:
Actually, see See contrib\src\microdns\0001-build-fix-getifaddrs-detection.patch in VLC
@tguillem can you answer this?

Reply:
Oh, VLC patches should be merged in this project, I'll do a PR.
But I don't see the link with Hugo's comment ?

Reply:
OK, these patches are already merged but we didn't do any new release since. That is why there are still in VLC.

Reply:
OK cool, thanks for checking.

Reply:
@tguillem oops - I copied the wrong VLC patch - it should be contrib\src\microdns\0002-fix-dummy-mdns_list_interfaces.patch. 

Rather than doing memset(pp_ip_addresses, 0, sizeof(char*)); do *pp_ip_addresses = NULL; 

But a few lines above it uses memset so I just followed the same logic to be consistent.

In your patch you changed it to memset:
-        **pp_intfs = NULL;
+        memset(intfs, 0, sizeof(*intfs));